bot/utils/persist.py

```diff
@@ -0,0 +1,40 @@
+import sqlite3
+from pathlib import Path
+from shutil import copyfile
+
+DIRECTORY = Path("data")  # directory that has a persistent volume mapped to it
+
+
+def datafile(file_path: Path) -> Path:
+    """
+    Copy datafile at the provided file_path to the persistent data directory.
+
+    A persistent data file is needed by some features in order to not lose data
+    after bot rebuilds.
+
+    This function will ensure that a clean data file with default schema,
+    structure or data is copied over to the persistent volume before returning
+    the path to this new persistent version of the file.
+
+    If the persistent file already exists, it won't be overwritten with the
+    clean default file, just returning the Path instead to the existing file.
+
+    Example Usage:
+    >>> clean_default_datafile = Path("bot", "resources", "datafile.json")
+    >>> persistent_file_path = datafile(clean_default_datafile)
+    """
+    if not file_path.exists():
+        raise OSError(f"File not found at {file_path}.")
+
+    persistant_path = Path(DIRECTORY, file_path.name)
+
+    if not persistant_path.exists():
```

Comment:
Could this lead to a scenario where the path exists as a directory, but a file with an identical path is trying to be added? Thus it won't be added even though it technically does not exist. Edge case I guess...

Reply:
Definitely edge case since filetypes should have an extension present while directories won't. However, I think I'll actually organise these datafiles in the persistent directory properly into subdirectories to avoid the possibility of two features each having a datafile with the same name and conflicting in that manner and to keep things cleaner. At the moment it just dumps all the datafiles in the one directory.